src/App.js

```diff
@@ -22,109 +22,98 @@ mapboxgl.accessToken = config.accessToken;
+            pitch,
+            interactive: false,
+        });
 
-    useEffect(() => {
         map.current.on('load', () => {
-            map.current.addSource('lineSource', {
-                type: 'geojson',
-                data: feature
-            });
-
-            map.current.addSource('pointSource', {
-                type: 'geojson',
-                data: feature
-            });
-
-            map.current.addSource('pointSourceBg', {
-                type: 'geojson',
-                data: feature
-            });
-
-            map.current.addLayer(config.layers.line);
-            map.current.addLayer(config.layers.circleBg, 'markers');
-            map.current.addLayer(config.layers.circle, 'markers');
-
-            const mapContainerSize = mapContainer.current.getBoundingClientRect();
- 
-            let mapOffset = 0;
-            
-            if (mapContainerSize.width >= 551 && mapContainerSize.width <= 1440 ) {
-                mapOffset = 0.28;
-            }
-            if (mapContainerSize.width > 1440) {
-                mapOffset = 0.3;
-            }
-
-            map.current.setPadding({left: mapContainerSize.width * mapOffset});
-           
-            scroller
-                .setup({
-                    step: '.step',
-                    offset: 0.5,
-                    progress: true
-                })
-                .onStepEnter((response) => actions.enter(response, config, map.current))
-                .onStepProgress((response) => actions.progress(response, config, map.current, data))
-                .onStepExit((response) => actions.exit(response));
-            
-            // bottom of page is reached, zoom out to entire route
-            window.onscroll = (ev) => {
-                if ((window.innerHeight + window.scrollY) >= document.body.offsetHeight) {
-                    map.current.setPitch(0);
-                    map.current.fitBounds([
-                        [
-                            7.411651611328124,
-                            46.93291653811045
-                        ],
-                        [
-                            7.922859191894531,
-                            47.0284823920254
-                        ]
-                    ], {
-                    padding: {left: mapContainerSize.width * mapOffset},
-                    bearing: 76
-                });
-                }
-            };
-
+            // This is the final time we need to reset scrollama
+            clearInterval(scrollReset);
+            resetScrollama();
         });
     });
 
+    // Set map padding
+    useEffect(() => {
+        if (!map.current) return;
+
+        const { width } = mapContainer.current.getBoundingClientRect();
+        const isBig = width > 1440;
+        const isMedium = !isBig && width > 550;
+        const mapOffset = isBig ? 0.3 : isMedium ? 0.28 : 0;
+        const left = width * mapOffset
+        map.current.setPadding({ left });
+    });
+
```

Comment:
lol this was such a shameful part of the code, thanks for making this look better 💚